tests/kokkos-based/hermitian_matrix_rank2_update_kokkos.cpp

```diff
@@ -0,0 +1,330 @@
+ /*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 2.0
+//              Copyright (2019) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "gtest_fixtures.hpp"
+#include "helpers.hpp"
+
+namespace{
+
+////////////////////////////////////////////////////////////
+// Utilities
+////////////////////////////////////////////////////////////
+
+// create rank-1 mdspan (vector)
+template <typename value_type,
```

Comment:
It seems like these functions are being repeated in every file.  Would you consider factoring them out into a separate header?  If so, then would you consider the following changes:

1. removing `inline`, as it is superfluous in this case; and
2. using the pointer type conversion function defined elsewhere (I think via PR #162) to `reinterpret_cast` `value_type` from `std::complex` to `Kokkos::complex` if needed?

Reply:
With pleasure - I hate duplication myself!
Please see PR https://github.com/kokkos/stdBLAS/pull/187 that reconciles test utilities and fixes `inline`.

Reply:
Ad.2. Where exactly would you see the cast to `Kokkos::complex` ?
Here `make_mdspan()` creates `mdspan` (which usually carries `std::complex`) and in the mapper we create `Kokkos::View` (which carries `Kokkos::complex`).

Reply:
Right, but that's `mdspan` to `View` conversion, so I'm missing out on where would you like me to cast to `Kokkos::complex` ? Here in the utilities we create and use `mdspan`s - except of [`parallel_reduce` in `is_same_matrix`](https://github.com/NexGenAnalytics/stdBLAS/blob/173_refactor_conj-if-needed/tests/kokkos-based/matrix_rank1_update_kokkos.cpp#L187-L196) which already uses `mdspan_to_view()` to prforming the conversions under consideration.

So where would you like this conversion and `Kokkos::complex` to appear ?
Your comment is attached to `make_mdspan()`, but it creates `mdspan` carrying `std::complex`, not `View` carrying `Kokkos::complex` ?

Reply:
Seems like this discussion is aimed at https://github.com/kokkos/stdBLAS/pull/187, which was already merged ?
@mhoemmen Please let me know if you'd wish any changes on this.